### PR TITLE
Feature: Debounce search button

### PR DESCRIPTION
**Pull request by andrewfluck**
See: #18

### CODE REVIEW

File: src/components/App.jsx [hunk 1]
```diff
@@ -71,18 +72,22 @@ class App extends React.Component {
...
+      event.preventDefault();
+
+    if (!this.state.isFetching && !this.state.buttonLock) {
+      this.setState({ isEmpty: true, isFetching: true });
+      const finalUrl = this.createUrl();
+      await fetch(finalUrl)
+        .then(this.handleErrors)
+        .then(resJson => {
+          this.setState({ isEmpty: false, isFetching: false, buttonLock: true, results: resJson, url: finalUrl });
+          setTimeout(() => this.setState({ buttonLock: false }), 5000);
```
Comment (trybick) [inline src/components/App.jsx]: Could we put the `setTimeout` as a callback to `setState`? Something like

```
this.setState({ isEmpty: false, isFetching: false, buttonLock: true, results: resJson, url: finalUrl }), () => {
      setTimeout(() => {
        this.setState({ buttonLock: false })
      }, 5000);
    });
  }
```

File: src/components/App.jsx [hunk 2]
```diff
@@ -71,18 +72,22 @@ class App extends React.Component {
...
-      .then(resJson => {
-        this.setState({ isEmpty: false, isFetching: false, results: resJson, url: finalUrl });
-      })
-      .catch(err => {
-        console.error('error:', err);
-        this.setState({ fetchError: true, isFetching: false });
-      });
+      event.preventDefault();
+
+    if (!this.state.isFetching && !this.state.buttonLock) {
```
Comment (trybick) [inline src/components/App.jsx]: Do you mind de-structuring these? Just want to keep it consistent with the rest of the project.

File: src/components/App.jsx [hunk 3]
```diff
@@ -183,6 +188,7 @@ class App extends React.Component {
           className="get-issues-btn"
           classNameWrapper="get-button-wrapper"
           forForm="issues-form"
+          disabled={this.state.isFetching || this.state.buttonLock}
```
Comment (trybick) [inline src/components/App.jsx]: de-structure `this.state`

**Comment by trybick** [comment]
Looks awesome, thanks @andrewfluck ! I left a few comments

File: src/components/App.jsx (hunk 1 above)
Comment (andrewfluck) [reply src/components/App.jsx]: Yeah, certainly, forgot about that little feature

File: src/components/App.jsx (hunk 2 above)
Comment (andrewfluck) [reply src/components/App.jsx]: oops, hoped you wouldn't notice any formatting issues, was using vim, and I didnt feel like changing the tab rule, so I tried fixing anything manually.

File: src/components/App.jsx (hunk 3 above)
Comment (andrewfluck) [reply src/components/App.jsx]: 👍🏻

File: src/components/App.jsx (hunk 2 above)
Comment (andrewfluck) [reply src/components/App.jsx]: oh, you didnt notice it, misread it, but I caught a formatting issue anyway

**Comment by trybick** [comment]
@andrewfluck Would you mind rebasing with master? While you're there we could we rename `buttonLock` to something like `isButtonDisabled`. ty!

**Comment by andrewfluck** [comment]
sure thing

**Comment by trybick** [comment]
There's still a merge conflict, is it possible to fix? @andrewfluck

**Comment by andrewfluck** [comment]
not sure, I think you have to since you have write access

**Comment by trybick** [comment]
@andrewfluck If you want to fix the merge conflicts or open a new PR with the debounce please do so. Otherwise I will plan to close this and just add your changes in manually.

**Comment by trybick** [comment]
Ty @andrewfluck - sorry it didn't work out merging

https://github.com/trybick/issue-collab/commit/2c8fafb9efd85ce0dd8be455d694d717d42bdf02

**Pull request closed without merging**